samx2x: separate RAM/Flash sizes by model.

Flameeyes opened the pull request:
samx2x: separate RAM/Flash sizes by model.



This creates separate dtsi files for the various memory density codes of
SAM X2xfamilies (they are the same where the specific size exists.)

All of the boards with the exclusion of EV11L78A use the same density
model of 18 (32KiB RAM and 256KiB flash) which is what the samd2x.dtsi
include specified for all of them previously.

The density code has been confirmed being the same across the D20/D21,
C20/C21, L21, and R21 families. This does not carry over to some other
series such as the E5x.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

nandojve commented on boards/arm/atsamr21_xpro/atsamr21_xpro.dts:
```diff
@@ -6,6 +6,7 @@
 
 /dts-v1/;
 #include <atmel/samr21.dtsi>
+#include <atmel/samd2xx18.dtsi>
```
The nomal is include `#include <atmel/samr21.dtsi>` inside `#include <atmel/samr2xx18.dtsi>`. This is vaiid for all models.

nandojve replied:
I was looking on all series and I'll propose that we add samx2x14 ... samx2x18 to define the memory. This way it will fit for all sam0 series.

Flameeyes replied:
Done that now, we're now down one file compared to the start, as it also covered the C20/C21 family.

Flameeyes committed:
samx2x: separate RAM/Flash sizes by model.



This creates separate dtsi files for the various memory density codes of
SAM X2xfamilies (they are the same where the specific size exists.)

All of the boards with the exclusion of EV11L78A use the same density
model of 18 (32KiB RAM and 256KiB flash) which is what the samd2x.dtsi
include specified for all of them previously.

The density code has been confirmed being the same across the D20/D21,
C20/C21, L21, and R21 families. This does not carry over to some other
series such as the E5x.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>
